### PR TITLE
[Feature] Adding support for egohands (hand segmentation) dataset 

### PR DESCRIPTION
**Pull request by navneeth**
## Motivation

Adding support for the [ego hands](http://vision.soic.indiana.edu/projects/egohands/) dataset. 
The dataset is introduced in this paper:
@InProceedings{Bambach_2015_ICCV,
author = {Bambach, Sven and Lee, Stefan and Crandall, David J. and Yu, Chen},
title = {Lending A Hand: Detecting Hands and Recognizing Activities in Complex Egocentric Interactions},
booktitle = {The IEEE International Conference on Computer Vision (ICCV)},
month = {December},
year = {2015}
}

This PR includes:

1. utility to import data : tools/convert_datasets/egohands.py
2. dataset implementation: mmseg/datasets/egohands.py
3. dataset config file: configs/_base_/datasets/egohands.py4. 

## Modification

1. utility to import data : tools/convert_datasets/egohands.py
2. dataset implementation: mmseg/datasets/egohands.py
3. dataset config file: configs/_base_/datasets/egohands.py4. 

## BC-breaking (Optional)

no breaking changes.

## Use cases (Optional)

hand segmentation

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
5. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
6. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
7. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by MeowZheng — COMMENTED** [review]
Many thanks for your contribution, we are working on doing experiments for testing `pspnet_unet_s5-d16_128x128_40k_egohands.py`

Besides, It would be better to add unit test for `mmseg/datasets/egohands.py`, please refer to https://github.com/open-mmlab/mmsegmentation/blob/master/tests/test_data/test_dataset.py

File: docs/en/dataset_prepare.md [hunk 1]
```diff
@@ -376,3 +383,16 @@ python tools/convert_datasets/isaid.py /path/to/iSAID
 ```
 
 In our default setting (`patch_width`=896, `patch_height`=896,　`overlap_area`=384), it will generate 33978 images for training and 11644 images for validation.
+
+### EGOHANDS
+
+The training and validation set of EGOHANDS could be download from [here](https://staffnet.kingston.ac.uk/~ku15565/CHASE_DB1/assets/CHASEDB1.zip).
```
Comment (MeowZheng) [inline docs/en/dataset_prepare.md:389]: please pay attention to this URL, it might not be for egohands?

Comment (navneeth) [reply docs/en/dataset_prepare.md]: you're right. Just corrected it.

File: configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+_base_ = [
+    '../_base_/models/pspnet_unet_s5-d16.py',
+    '../_base_/datasets/egohands.py', '../_base_/default_runtime.py',
+    '../_base_/schedules/schedule_40k.py'
+]
+model = dict(test_cfg=dict(crop_size=(128, 128), stride=(85, 85)))
```
Comment (MeowZheng) [inline configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: I have a little question about the `num_classes` of model, does it require modification?

File: configs/_base_/datasets/egohands.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            dict(type='Resize', keep_ratio=True),
+            dict(type='RandomFlip'),
+            dict(type='Normalize', **img_norm_cfg),
+            dict(type='ImageToTensor', keys=['img']),
+            dict(type='Collect', keys=['img'])
+        ])
+]
+
+data = dict(
+    samples_per_gpu=4,
```
Comment (MeowZheng) [inline configs/_base_/datasets/egohands.py:37]: another question is num of GPUs,  `samples_per_gpu` is 4 but how many GPUs are used for training?

**Comment by navneeth** [comment]
> 
> Besides, It would be better to add unit test for `mmseg/datasets/egohands.py`, please refer to https://github.com/open-mmlab/mmsegmentation/blob/master/tests/test_data/test_dataset.py

Just added one. Please check it.

File: configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py (hunk 2 above)
Comment (navneeth) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: `
     CLASSES = ('background', 'hand1_lt', 'hand1_rt', 'hand2_lt', 'hand2_rt')
`
Should be handled by this code in: mmseg/datasets/egohands.py

Comment (MeowZheng) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: `num_classes` is the model hyperparameter, which decides the lastest layer output channels of the model.

If egohand dataset has 5 classes, I think should modify the `num_classes=5` of model.

Comment (navneeth) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: How is the best way to do that ? Should I set:
`model[auxiliary_head][num_classes] =5`

Comment (MeowZheng) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: I have run the current training configuration, and `num_classes` maybe not quite right.

Please confirm that the current model training configuration is correct

Comment (navneeth) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: Please try training with this config configs/fcn/fcn_r50-d8_512x1024_40k_egohands.py . I can confirm that it works for me.

Comment (MeowZheng) [reply configs/unet/pspnet_unet_s5-d16_128x128_40k_egohands.py:6]: OK I'll try it later

**Comment by OpenMMLab-Assistant005** [comment]
Hi @navneeth ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
